### PR TITLE
Big JS refactor

### PR DESCRIPTION
**Pull request by eadpearce**

### CODE REVIEW

File: caseworker/templates/layouts/base.html [hunk 1]
```diff
@@ -137,48 +140,17 @@
...
 	<script src="{% static 'javascripts/back-link.js' %}"></script>
+	<script src="{% static 'javascripts/menu-tooltips.js' %}"></script>
 
 	<script nonce="{{ request.csp_nonce }}">
 		window.GOVUKFrontend.initAll()
 	</script>
 
 	{% block javascript %}{% endblock %}
 
-	<script nonce="{{ request.csp_nonce }}" type="text/javascript">
```
Comment (eadpearce) [inline caseworker/templates/layouts/base.html:149]: this is now in menu-tooltips.js

File: caseworker/templates/queues/cases.html [hunk 2]
```diff
@@ -124,170 +81,30 @@ <h1 class="govuk-heading-l">
...
-				}
-			});
+{% endblock %}
 
-			if (!anyVisible) {
-				$("#queues > span").show()
-			}
-		});
+{% block javascript %}
+<div class="app-hidden--force" id="js-chevron-svg">
```
Comment (eadpearce) [inline caseworker/templates/queues/cases.html:99]: this is to get around having to write the JS in the template as we append this chevron icon to the page in flag-expanders.js

File: caseworker/templates/queues/cases.html [hunk 3]
```diff
@@ -124,170 +81,30 @@ <h1 class="govuk-heading-l">
...
+	{% svg 'chevron' %}
+</div>
 
-		$("#link-queue").removeAttr("href");
+	<script nonce="{{ request.csp_nonce }}" src="{% static 'main.js' %}"></script>
 
-		function generateQueuesMenu() {
```
Comment (eadpearce) [inline caseworker/templates/queues/cases.html:274]: this has moved to queues-menu.js

File: caseworker/core/constants.py [hunk 4]
```diff
@@ -6,6 +8,10 @@
 
 ENFORCEMENT_XML_MAX_FILE_SIZE = 1000000  # 1 MB
```
Comment (saruniitr) [inline caseworker/core/constants.py:9]: Not your change but this is not correct, should be `1024 * 1024`

File: caseworker/queues/views.py [hunk 5]
```diff
@@ -27,31 +34,69 @@
...
+    def queue_pk(self):
+        return self.kwargs.get("queue_pk") or self.request.session.get("default_queue")
+
+    @cached_property
+    def data(self):
+        hidden = self.request.GET.get('hidden')
+
+        params = {'page': int(self.request.GET.get('page', 1))}
+        for key, value in self.request.GET.items():
+            if key != 'flags[]':
```
[the remark this answers is not in the document]
Comment (eadpearce) [reply caseworker/queues/views.py:60]: I have no idea actually. I didn't really understand this code and didn't want to mess with it so just copied it into the new view class

File: core/builtins/custom_tags.py [hunk 6]
```diff
@@ -37,6 +38,43 @@
 STRING_NOT_FOUND_ERROR = "STRING_NOT_FOUND"
 
 
+@register.filter(name="sla_colour")
+def sla_colour(value, arg):
+    """
+    Template tag to calculate the colour of caseworker/templates/includes/sla_display.html
+    """
+    since_raised = int(value)
```
[the remark this answers is not in the document]
Comment (eadpearce) [reply core/builtins/custom_tags.py:46]: That would be nice! But I think this is how it was designed to work. I don't understand it fully. Might be worth raising with Daniel for a bit more clarity on what these are for. Personally I'd like more descriptive naming than just the SLA acronym

File: caseworker/queues/views.py [hunk 7]
```diff
@@ -27,31 +34,69 @@
...
+
+    @property
+    def queue_pk(self):
+        return self.kwargs.get("queue_pk") or self.request.session.get("default_queue")
+
+    @cached_property
+    def data(self):
+        hidden = self.request.GET.get('hidden')
+
+        params = {'page': int(self.request.GET.get('page', 1))}
```
[the remark this answers is not in the document]
Comment (eadpearce) [reply caseworker/queues/views.py:58]: We have questionable formatting throughout the codebase unfortunately. But I can run black on the files I've touched at least. Maybe eventually we'll get through everything lol

File: caseworker/queues/views.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (eadpearce) [reply caseworker/queues/views.py]: Okay I figured out why this is like this and spoke with @richtier to get a fix for it as this code here is pretty confusing. Will push that shortly

[the remark this answers is not in the document]
Comment (eadpearce) [reply caseworker/queues/views.py]: Re: discussion on slack - I think I will leave tackling this for later as it turns out it's a bigger job than I first thought.

**Review by richtier — APPROVED** [review]

**Pull request merged**